pymongo/mongo_client.py

```diff
@@ -1507,65 +1472,52 @@ def _retryable_read(
+        retryable = bool(
             retryable and self.options.retry_reads and not (session and session.in_transaction)
         )
-        last_error: Optional[Exception] = None
-        retrying = False
-        multiple_retries = _csot.get_timeout() is not None
-
-        while True:
-            if retrying:
-                remaining = _csot.remaining()
-                if remaining is not None and remaining <= 0:
-                    assert last_error is not None
-                    raise last_error
-            try:
-                server = self._select_server(read_pref, session, address=address)
-                with self._conn_from_server(read_pref, server, session) as (
-                    conn,
-                    read_pref,
-                ):
-                    if retrying and not retryable:
-                        # A retry is not possible because this server does
-                        # not support retryable reads, raise the last error.
-                        assert last_error is not None
-                        raise last_error
-                    return func(session, server, conn, read_pref)
-            except ServerSelectionTimeoutError:
-                if retrying:
-                    # The application may think the write was never attempted
-                    # if we raise ServerSelectionTimeoutError on the retry
-                    # attempt. Raise the original exception instead.
-                    assert last_error is not None
-                    raise last_error
-                # A ServerSelectionTimeoutError error indicates that there may
-                # be a persistent outage. Attempting to retry in this case will
-                # most likely be a waste of time.
-                raise
-            except ConnectionFailure as exc:
-                if not retryable or (retrying and not multiple_retries):
-                    raise
-                retrying = True
-                last_error = exc
-            except OperationFailure as exc:
-                if not retryable or (retrying and not multiple_retries):
-                    raise
-                if exc.code not in helpers._RETRYABLE_ERROR_CODES:
-                    raise
-                retrying = True
-                last_error = exc
+        return self._retry_internal(
+            func,
+            session,
+            None,
+            is_read=True,
+            address=address,
+            read_pref=read_pref,
+            retryable=retryable,
+        )
 
     def _retryable_write(
         self,
         retryable: bool,
-        func: Callable[[Optional[ClientSession], Connection, bool], T],
+        func: _WriteCall[T],
         session: Optional[ClientSession],
+        bulk: Optional[_Bulk] = None,
     ) -> T:
-        """Internal retryable write helper."""
+        """Execute an operation with at most one consecutive retries on write
```

Comment:
Won't there be more than one attempt if `csot` is enabled?

Reply:
Admittedly, I did not assume any retries occurred more than once. I grifted the docstring from [`_retry_with_session`](https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/mongo_client.py#L1403-L1408) which led me to assume retry was only a "one and done" scenario. 

Thanks for raising the question, though. I ended up reviewing the `_retry_internal` call--you're definitely right. If `csot` is enabled, we check for the `_csot.get_timeout()` and that becomes the determinant for retries.

I'll go ahead and change _all_ the docstrings to reflect that.